### PR TITLE
Short channel matching for julia command

### LINKED ISSUE #830: Support short channel matching
On julia you can do `--start=no` to control `--startup-file=no` because there's no other arg that matches `--start`

It would be nice if juliaup did the same for channel names

So `julia +n` for nightly. `julia +r` for release. etc.

### PR DESCRIPTION
**Pull request by rakeshksr**
Closes #830
This pull request introduces the feature of short channel matching for the Julia command. Users can now launch alpha, beta, LTS, release, and nightly versions by specifying their respective short names: "a" for alpha, "b" for beta, "l" for LTS, "r" for release, and "n" for nightly.

### CODE REVIEW

**Comment by LilithHafner** [comment]
This error message is interesting, but upon reflection seems pretty good. It would be slightly better if the message explicitly mentioned that `l` expanded to `lts`.

```
x@fedora:~/.julia/dev/juliaup$ target/debug/julia +l
`lts` is not installed. Please run `juliaup add lts` to install channel or version.
```

File: src/bin/julialauncher.rs [hunk 1]
```diff
@@ -305,7 +305,14 @@ fn run_app() -> Result<i32> {
...
         if let Some(stripped) = first_arg.strip_prefix('+') {
-            channel_from_cmd_line = Some(stripped.to_string());
+            channel_from_cmd_line = match stripped {
+                "a" => Some("alpha".to_string()),
+                "b" => Some("beta".to_string()),
+                "l" => Some("lts".to_string()),
+                "n" => Some("nightly".to_string()),
+                "r" => Some("release".to_string()),
+                strip => Some(strip.to_string()),
+            }
```
Comment (LilithHafner) [inline src/bin/julialauncher.rs:315]:
```suggestion
            channel_from_cmd_line = Some(match stripped {
                "a" => "alpha".to_string(),
                "b" => "beta".to_string(),
                "l" => "lts".to_string(),
                "n" => "nightly".to_string(),
                "r" => "release".to_string(),
                strip => strip.to_string(),
            })
```
Factor out the `Some`

File: src/bin/julialauncher.rs [hunk 2]
```diff
@@ -305,7 +305,14 @@ fn run_app() -> Result<i32> {
...
         if let Some(stripped) = first_arg.strip_prefix('+') {
-            channel_from_cmd_line = Some(stripped.to_string());
+            channel_from_cmd_line = Some(match stripped {
+                "a" => "alpha".to_string(),
+                "b" => "beta".to_string(),
+                "l" => "lts".to_string(),
+                "n" => "nightly".to_string(),
+                "r" => "release".to_string(),
+                strip => strip.to_string(),
+            })
```
Comment (LilithHafner) [inline src/bin/julialauncher.rs:315]:
```suggestion
            channel_from_cmd_line = Some(match stripped {
                "a" => "alpha",
                "b" => "beta",
                "l" => "lts",
                "n" => "nightly",
                "r" => "release",
                strip => strip,
            }.to_string())
```
Factor out to_string()

File: src/bin/julialauncher.rs [hunk 3]
```diff
@@ -305,7 +305,14 @@ fn run_app() -> Result<i32> {
...
         if let Some(stripped) = first_arg.strip_prefix('+') {
-            channel_from_cmd_line = Some(stripped.to_string());
+            channel_from_cmd_line = Some(match stripped {
+                "a" => "alpha",
+                "b" => "beta",
+                "l" => "lts",
+                "n" => "nightly",
+                "r" => "release",
+                strip => strip,
+            }.to_string())
```
Comment (LilithHafner) [inline src/bin/julialauncher.rs:315]:
```suggestion
            channel_from_cmd_line = Some(
                match stripped {
                    "a" => "alpha",
                    "b" => "beta",
                    "l" => "lts",
                    "n" => "nightly",
                    "r" => "release",
                    strip => strip,
                }
                .to_string()
            )
```

File: src/bin/julialauncher.rs [hunk 4]
```diff
@@ -305,7 +305,17 @@ fn run_app() -> Result<i32> {
...
+            channel_from_cmd_line = Some(
+                match stripped {
+                    "a" => "alpha",
+                    "b" => "beta",
+                    "l" => "lts",
+                    "n" => "nightly",
+                    "r" => "release",
+                    strip => strip,
+                }
+                .to_string()
```
Comment (LilithHafner) [inline src/bin/julialauncher.rs:317]:
```suggestion
                .to_string(),
```

**Review by LilithHafner — APPROVED** [review]
This only supports single character abbreviations, not intermediary abbreviations. I think that's okay.

LGTM

**Comment by LilithHafner** [comment]
Would be good to add a test, too

**Comment by LilithHafner** [comment]
Needs docs

**Comment by davidanthoff** [comment]
I think I've finally am getting clarity what I'm thinking about this, see https://github.com/JuliaLang/juliaup/issues/830.

**Comment by davidanthoff** [comment]
I'm closing this for now, as I don't think we will go with this kind of solution.

**Pull request closed without merging**